lld/COFF/InputFiles.h

```diff
@@ -403,8 +404,8 @@ class BitcodeFile : public InputFile {
 // .dll file. MinGW only.
 class DLLFile : public InputFile {
 public:
-  explicit DLLFile(COFFLinkerContext &ctx, MemoryBufferRef m)
-      : InputFile(ctx, DLLKind, m) {}
+  explicit DLLFile(SymbolTable &symtab, MemoryBufferRef m)
```

Comment:
Here in `DLLFile`, we change the constructor to take a `SymbolTable&` instead, while for `ObjFile` above, we keep taking a `COFFLinkerContext&` parameter, and resolve `ctx.symtab` within the `ObjFile` constructor; 1) why the inconsistency, and 2) shouldn't we stop doing `ctx.symtab` here? Or is that a leftover that gets taken care of in one of the upcoming patches?

Reply:
I'll change it to `COFFLinkerContext`. Importing DLLs is more complicated on ARM64X because those DLLs can themselves be ARM64X files. If that's the case, we need to detect it and create two `DLLFile` instances: one for the original memory buffer using the native namespace and another for the hybrid view obtained from `getHybridObjectView()`. With this mechanism, the caller will already know the exact namespace for the created file instance, so they might as well pass it. That said, we can probably address this later.

Reply:
Ah, I remember now why I didn't change it: the constructor would need to be moved out of the header because `COFFLinkerContext` is an incomplete type in `InputFiles.h`. Since it will ultimately use the `symtab` argument (as mentioned above), I updated the type in this commit already. Should I change it?

Reply:
I don't really have a strong opinion on it; moving the constructor to the .cpp file here adds a bit more churn if it can be moved back to the header later, so it's probably fine to keep it like this for now - I mostly wanted to raise the issue and hear the reasoning around it.